src/index.ts

```diff
@@ -64,45 +119,68 @@ export const findNewDependencies = packageDiff => {
   return added
 }
 
-export const getYarnMetadataForDep = async dep => {
-  return new Promise<string>(resolve => {
-    child_process.exec(`yarn why '${dep}' --json`, (err, output) => {
-      if (output) {
-        // Comes as a series of little JSON messages
-        const usefulJSONContents = output.toString().split(`{"type":"activityEnd","data":{"id":0}}`).pop() as string
-        const asJSON = usefulJSONContents.split("}\n{").join("},{")
-
-        const whyJSON = JSON.parse(`[${asJSON}]`)
-        const messages = whyJSON.filter(msg => typeof msg.data === "string").map(m => m.data)
-        resolve(`
-  <details>
-    <summary><code>yarn why ${printDep(dep)}</code> output</summary>
-    <ul><li><code>${messages.join("</code></li><li><code>")}
-    </code></li></ul>
-  </details>
-  `)
-      } else {
-        resolve("")
-      }
-    })
-  })
+export interface YarnWhyMessage {
+  type: "activityStart" | "activityTick" | "activityEnd" | "step" | "info"
+  data: object | string
 }
 
-const safeLink = (name: string) => `<a href='${linkToNPM(name)}'><code>${printDep(name)}</code></a>`
-const printDep = (name: string) => name.replace(/@/, "&#64;")
-const linkToNPM = (name: string) => `https://www.npmjs.com/package/${name}`
+export const getYarnMetadataForDep = async (
+  dep: string,
+  { renderYarnWhy = defaultRenderYarnWhy }: Options = {},
+): Promise<string> => {
+  const { stdout } = await exec(`yarn why '${dep}' --json`)
+
+  if (stdout) {
+    // Comes as a series of little JSON messages
+    const whyMessages: ReadonlyArray<YarnWhyMessage> = stdout.split("\n").map(line => JSON.parse(line.trim()))
```

Comment:
I changed your logic to not string-replace it into a different JSON structure but instead to split on new line and parse each of the lines as a separate piece of JSON. Maybe slower but I think it's simpler...

Reply:
I'm fine with that, I was probably going a bit far anyway